bcda/api/requests_test.go

```diff
@@ -0,0 +1,150 @@
+package api
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/CMSgov/bcda-app/bcda/auth"
+	"github.com/CMSgov/bcda-app/bcda/database"
+	"github.com/CMSgov/bcda-app/bcda/models"
+	"github.com/bgentry/que-go"
+	"github.com/jackc/pgx"
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/go-chi/chi"
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	acoID = "DBBD1CE1-AE24-435C-807D-ED45953077D3"
+)
+
+type RequestsTestSuite struct {
+	suite.Suite
+
+	runoutEnabledEnvVar string
+	origSvc             models.Service
+
+	db *gorm.DB
+}
+
+func TestRequestsTestSuite(t *testing.T) {
+	suite.Run(t, new(RequestsTestSuite))
+}
+
+func (s *RequestsTestSuite) SetupSuite() {
+	s.db = database.GetGORMDbConnection()
+
+	queueDatabaseURL := os.Getenv("QUEUE_DATABASE_URL")
+	pgxcfg, err := pgx.ParseURI(queueDatabaseURL)
+	if err != nil {
+		s.T().Error(err)
+	}
+
+	pgxpool, err := pgx.NewConnPool(pgx.ConnPoolConfig{
+		ConnConfig:   pgxcfg,
+		AfterConnect: que.PrepareStatements,
+	})
+	if err != nil {
+		s.T().Error(err)
+	}
+
+	qc := que.NewClient(pgxpool)
+	SetQC(qc)
+}
+
+func (s *RequestsTestSuite) SetupTest() {
+	s.runoutEnabledEnvVar = os.Getenv("BCDA_ENABLE_RUNOUT")
+	s.origSvc = svc
+}
+
+func (s *RequestsTestSuite) TearDownSuite() {
+	s.db.Close()
+}
+
+func (s *RequestsTestSuite) TearDownTest() {
+	os.Setenv("BCDA_ENABLE_RUNOUT", s.runoutEnabledEnvVar)
+	svc = s.origSvc
+}
+
+func (s *RequestsTestSuite) TestRunoutEnabled() {
+	os.Setenv("BCDA_ENABLE_RUNOUT", "true")
+	qj := []*que.Job{&que.Job{Type: "ProcessJob"}, &que.Job{Type: "ProcessJob"}}
+	tests := []struct {
+		name string
+
+		errToReturn error
+		respCode    int
+	}{
+		{"Successful", nil, http.StatusAccepted},
+		{"No CCLF file found", models.CCLFNotFoundError{}, http.StatusNotFound},
+		{"Some other error", errors.New("Some other error"), http.StatusInternalServerError},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(t *testing.T) {
+			mockSvc := &models.MockService{}
+			var jobs []*que.Job
+			if tt.errToReturn == nil {
+				jobs = qj
+			}
+
+			mockSvc.On("GetQueJobs", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(jobs, tt.errToReturn)
```

Comment:
By using the mocks here, we can test the behavior of our request handler in various situations. 

Instead of ensuring that we DO NOT have a CCLF8 runout file for our test ACO, we can simply mock the response of our service interface. We do not have to stage all of this data in our postgres database.